src/triage/component/catwalk/storage.py

```diff
@@ -213,123 +243,179 @@ def matrix_with_sorted_columns(self, columns):
         # when we serialize (say, for multiprocessing),
         # we don't want the cached members to show up
         # as they can be enormous
-        self._matrix = None
+        self.matrix = None
         self._labels = None
-        self._metadata = None
-        self._head_of_matrix = None
+        self.metadata = None
+        self.head_of_matrix = None
         return self.__dict__.copy()
 
 
 class HDFMatrixStore(MatrixStore):
 
+    def __init__(self, matrix_path=None, metadata_path=None):
+        super().__init__(matrix_path, metadata_path)
+        self.metadata = self.load_metadata()
+
     def _get_head_of_matrix(self):
         try:
-            hdf = pandas.HDFStore(self.matrix_path)
-            key = hdf.keys()[0]
-            head_of_matrix = hdf.select(key, start=0, stop=1)
-            head_of_matrix.set_index(self.metadata['indices'], inplace=True)
-            self._head_of_matrix = head_of_matrix
-        except pandas.error.EmptyDataError:
-            self._head_of_matrix = None
+            head_of_matrix = pd.read_hdf(self.matrix_path, start=0, stop=1)
+            # Is the index already in place?
+            if head_of_matrix.index.name not in self.metadata['indices']:
+                head_of_matrix.set_index(self.metadata['indices'], inplace=True)
+        except pd.errors.EmptyDataError:
+            head_of_matrix = None
+
+        return head_of_matrix
 
     def _load(self):
-        with smart_open.smart_open(self.matrix_path, "rb") as f:
-            self._matrix = self._read_hdf_from_buffer(f)
-        self._metadata = self.load_yaml(self.metadata_path)
-        try:
-            self._matrix.set_index(self._metadata['indices'], inplace=True)
-        except Exception:
-            pass
-
-    def _read_hdf_from_buffer(self, buffer):
-        with pandas.HDFStore(
-                "data.h5",
-                mode="r",
-                driver="H5FD_CORE",
-                driver_core_backing_store=0,
-                driver_core_image=buffer.read()) as store:
-
-            if len(store.keys()) > 1:
-                raise Exception('Ambiguous matrix store. More than one dataframe in the hdf file.')
-
-            try:
-                return store["matrix"]
-
-            except KeyError:
-                print("The hdf file should contain one and only key, matrix.")
-                return store[store.keys()[0]]
-
-    def _write_hdf_to_buffer(self, df):
-        with pandas.HDFStore(
-                "data.h5",
-                mode="w",
-                driver="H5FD_CORE",
-                driver_core_backing_store=0) as out:
-            out["matrix"] = df
-            return out._handle.get_file_image()
+        path_parsed = urlparse(self.matrix_path)
+        scheme = path_parsed.scheme  # If '' of 'file' is a regular file or 's3'
+
+        if scheme in ('', 'file'):  # Local file
+            matrix = pd.read_hdf(self.matrix_path)
+        else:
+            raise ValueError(f"""
```

Comment:
What is the problem with s3 and HDF? I know Eddie ran into some problems but seemed to work past them.